web/src/client/storage.js

```diff
@@ -451,6 +451,43 @@ class ProposalManager {
     return proxy.AvailableDevices.map(path => findDevice(systemDevices, path)).filter(d => d);
   }
 
+  /**
+   * Gets the devices that can be selected as target for a volume.
+   *
+   * A device can be selected as target for a volume if either it is an available device for
+   * installation or it is a device built over the available devices for installation. For example,
+   * a MD RAID is a possible target only if all its members are available devices or children of the
+   * available devices.
+   *
+   * @returns {Promise<StorageDevice[]>}
+   */
+  async getVolumeDevices() {
+    /** @type {StorageDevice[]} */
+    const availableDevices = await this.getAvailableDevices();
+
+    /** @type {(device: StorageDevice) => boolean} */
+    const isAvailable = (device) => {
+      const isChildren = (device, parentDevice) => {
+        const partitions = parentDevice.partitionTable?.partitions || [];
+        return !!partitions.find(d => d.name === device.name);
+      };
+
+      return (
+        !!availableDevices.find(d => d.name === device.name) ||
+        !!availableDevices.find(d => isChildren(device, d))
```

Comment:
NP: I wonder why only one `find` is not enough.

Reply:
You are right. I guess it was at different places in some preliminary code and then it ends so.